telemetry/tracks/class-tracks-event.php

```diff
@@ -0,0 +1,336 @@
+	 * @param array<string, mixed>|array<empty> $event_properties Any event properties to be processed.
+	 * @return Tracks_Event_DTO The resulting event object with processed properties.
+	 */
+	protected static function process_properties(
+		string $event_prefix,
+		string $event_name,
+		array $event_properties
+	): Tracks_Event_DTO {
+		$event = static::encode_properties( $event_properties );
+		$event = static::set_user_properties( $event );
+
+		// Set event name. If the event name doesn't have the prefix, add it.
+		$event->_en = preg_replace(
+			'/^(?:' . $event_prefix . ')?(.+)/',
+			$event_prefix . '\1',
+			$event_name
+		) ?? '';
+
+		// Set event timestamp.
+		if ( ! isset( $event->_ts ) ) {
+			$event->_ts = static::milliseconds_since_epoch();
+		}
+
+		// Remove non-routable IPs to prevent record from being discarded.
+		if ( isset( $event->_via_ip ) &&
+			1 === preg_match( '/^192\.168|^10\./', $event->_via_ip ) ) {
+			unset( $event->_via_ip );
+		}
+
+		// Set VIP environment if it exists.
+		if ( defined( 'VIP_GO_APP_ENVIRONMENT' ) ) {
+			$app_environment = constant( 'VIP_GO_APP_ENVIRONMENT' );
+			if ( is_string( $app_environment ) && '' !== $app_environment ) {
+				$event->vipgo_env = $app_environment;
+			}
+		}
+
+		// Set VIP organization if it exists.
+		if ( defined( 'VIP_ORG_ID' ) ) {
+			$org_id = constant( 'VIP_ORG_ID' );
+			if ( is_string( $org_id ) && '' !== $org_id ) {
+				$event->vipgo_org = $org_id;
+			}
+		}
+
+		// Check if the user is a VIP user.
+		$event->is_vip_user = Support_User::user_has_vip_support_role( get_current_user_id() );
+
+		return $event;
+	}
+
+	/**
+	 * Sets the Tracks User ID and User ID Type depending on the current
+	 * environment.
+	 *
+	 * @param Tracks_Event_DTO $event The event to annotate with identity information.
+	 * @return Tracks_Event_DTO The new event object including identity information.
+	 */
+	protected static function set_user_properties( Tracks_Event_DTO $event ): Tracks_Event_DTO {
+		$wp_user = wp_get_current_user();
+
+		// Only track logged-in users.
+		if ( 0 === $wp_user->ID ) {
+			return $event;
+		}
+
+		// Set anonymized event user ID; it should be consistent across environments.
+		if ( defined( 'VIP_TELEMETRY_SALT' ) ) {
+			$salt           = constant( 'VIP_TELEMETRY_SALT' ); // this would be private but globally shared across the platform
+			$tracks_user_id = hash_hmac( 'sha256', $wp_user->user_email, $salt );
+
+			$event->_ui = $tracks_user_id;
+			$event->_ut = 'vip:user_email';
+
+			return $event;
+		}
+
+		// Users in the VIP environment.
+		if ( defined( 'VIP_GO_APP_ID' ) ) {
+			$app_id = constant( 'VIP_GO_APP_ID' );
+			if ( is_integer( $app_id ) && 0 < $app_id ) {
+				$event->_ui = sprintf( '%s_%s', $app_id, $wp_user->ID );
+				$event->_ut = 'vip_go_app_wp_user';
+
+				return $event;
+			}
+		}
+
+		// All other environments.
+		$wp_base_url = get_option( 'home' );
+		if ( ! is_string( $wp_base_url ) || '' === $wp_base_url ) {
```

Comment:
Under which conditions this would've matched?

Reply:
What would you like to see here instead?

It sounds like it could happen in case of a major fault, or deliberate malice. 

```
function set_home_option_empty( $value ) {
    return '';
}

add_filter( 'option_home', 'set_home_option_empty' );
```


For more context, this is from Parsely's implementation:

https://github.com/Automattic/vip-go-mu-plugins/blob/5bf617632fbcb65b95e1957a19139e1091f96470/vip-parsely/Telemetry/Tracks/class-tracks-event.php#L53-L56

Reply:
Since I don't have a strong opinion about this fallback (I'm following in the footsteps of the author of https://github.com/Automattic/vip-go-mu-plugins/pull/5770), I removed it. Please let me know if you want it back.